src/VisualStudio/Core/SolutionExplorerShim/HierarchyItemToProjectIdMap.cs

```diff
@@ -42,7 +43,9 @@ public bool TryGetProjectId(IVsHierarchyItem hierarchyItem, string targetFramewo
                     .Where(p =>
                     {
                         if (p.Hierarchy.TryGetCanonicalName((uint)VSConstants.VSITEMID.Root, out string projectCanonicalName)
-                            && projectCanonicalName.Equals(nestedCanonicalName, System.StringComparison.OrdinalIgnoreCase))
+                            && p.Hierarchy.TryGetItemName((uint)VSConstants.VSITEMID.Root, out string projectName)
+                            && projectCanonicalName.Equals(nestedCanonicalName, System.StringComparison.OrdinalIgnoreCase)
+                            && projectName.Equals(nestedName))
```

Comment:
@Pilchie I've checked the [__VSHPROPID](https://msdn.microsoft.com/en-us/library/microsoft.visualstudio.shell.interop.__vshpropid.aspx) enumeration (and derived enumerations) several times for such a property and never found anything appropriate.

I probably don't need `OrdinalIgnoreCase` for the canonical name, either, but I'm not eager to make this change any bigger than necessary.

I will document what I'm using here, and why.